src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/String/KeyAnalyzer.cs

```diff
@@ -32,31 +32,40 @@ internal static class KeyAnalyzer
             ReadOnlySpan<string> uniqueStrings, bool ignoreCase, int minLength, int maxLength)
         {
             Debug.Assert(!uniqueStrings.IsEmpty);
+            bool allUniqueStringsAreConfirmedAscii = ignoreCase && AreAllAscii(uniqueStrings);
```

Comment:
It seems like the name here should be like "confirmed ASCII and ignore case is set too"?

Reply:
If it's false, it doesn't mean it's not ASCII, it just means it wasn't confirmed to be. If it's true, it doesn't matter whether ignorecase was set or not.

Reply:
> If it's false, it doesn't mean it's not ASCII, it just means it wasn't confirmed to be.

Can `IsAllAscii` return false negatives?

Reply:
No